src/hooks/useEtherspotBalances.ts

```diff
@@ -0,0 +1,52 @@
+import { useEtherspot } from '@etherspot/react-etherspot';
+import { useEffect, useState } from 'react';
+import { AccountBalance, AccountTypes } from 'etherspot';
+
+let isConnecting: Promise<any> | undefined;
+
+/**
+ * Hook to fetch account balances
+ * @param chainId {number} - Chain ID
+ * @returns {AccountBalance[] | null} - Account balances
+ */
+const useEtherspotBalances = (chainId: number = 1): AccountBalance[] | null => {
+  const [balances, setBalances] = useState<AccountBalance[] | null>(null);
+  const { connect, getSdkForChainId } = useEtherspot();
+
+
+  useEffect(() => {
+    let shouldUpdate = true;
+
+    const updateBalance = async () => {
+      if (!!isConnecting) return;
+
+      const sdkForChainId = getSdkForChainId(chainId);
+      if (!sdkForChainId) return;
+
+      if (sdkForChainId?.state?.account?.type !== AccountTypes.Contract) {
+        isConnecting = connect(chainId);
+        await isConnecting;
+        isConnecting = undefined;
+      }
+
+      try {
+        const { items } = await sdkForChainId.getAccountBalances({
+          account: sdkForChainId.state.account.address,
+          chainId,
+        });
+        if (shouldUpdate) setBalances(items);
+      } catch (e) {
+        //
```

Comment:
We should log something here, at least a `console.warn` to show developers that there was a problem with the service as opposed to something that they have done

Reply:
added, thanks!